File: 01_Leetcode_FAANG/Pritam/Diagonal_sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        sum = 0
        for i in range(len(mat)):
            sum = sum + mat[i][i]
            mat[i][i] = 0
            sum = sum + mat[i][len(mat)-1-i]
        return sum

File: 01_Leetcode_FAANG/Bikash Dalai/diagonal_sum.cpp
class Solution {
public:
    int diagonalSum(vector<vector<int>>& mat) {
        int total=0;
        int n=mat.size();
        for(int i=0;i<mat.size();i++)
        {
            for(int j=0;j<mat.size();j++)
            {
                if((i+j)==n-1)
                {
                    cout<<i<<" "<<j<<endl;
                    total+=mat[i][j];
                    
                }
                else if(i==j)
                {
                    total+=mat[i][j];
                }
                
            }
        }
        
        return total;
    }
};
